pkg/git/git.go

```diff
@@ -160,9 +149,7 @@ func plumbCommand(cmd *exec.Cmd, channel ssh.Channel, sidechannel io.Writer) *sy
 	}()
 
 	cmd.Stdout = channel
-	cmd.Stderr = channel.Stderr()
-
-	return &wg
```

Comment:
>I take it the old waitgroup add calls were all removed at some intermediary point

It appears so, because `plumbCommand` didn't modify this `WaitGroup`, nor did any caller. The one caller of this `func` called `Wait` on the `WaitGroup`, which was effectively a no-op because there was never any `Add` call

>At one point, the purpose of the wait group was to make sure that for large inputs, we could read in the background, but parallelize calls at a higher level. It looks like this may have eroded away over time, but I'm concerned that a big input could have some weird issues if this copy is still going on at the same time that sidechannel gets consumed.

Makes sense, but where would we need to parallelize at a higher level? This and the calling `func Receive` is already running in a goroutine. Anyway, the `cmd.Wait` in the aforementioned `func Receive` will block progress before the sidechannel is consumed (or, in practical terms, `errbuf.Bytes()` is called). See anything I'm missing?

>More specifically, if this is 3/4 of the way through a write on the io.Copy operation, and you read to the end of the sidechannel file, will you get an EOF, or will it start blocking?

Excellent question. The `io.Copy` just copies all input from the `ssh.Channel` to the command's STDIN, so it won't have any effect on the sidechannel `io.Writer`. As soon as the `ssh.Channel` returns `EOF` from its `Read()`, the copy will return with no error and then stdin pipe will close. From testing, that doesn't seem to be an issue for the `git-shell` command. Could I be wrong?

However, there is a bug here because as soon as data starts coming in on STDIN to the command, it may start writing to STDOUT or STDERR, but we haven't assigned `io.Writer`s to `cmd.Stdout` or `cmd.Stderr`. I'll push a fix for that.